src/ComunicaEngine.js

```diff
@@ -12,17 +12,26 @@ export default class ComunicaEngine {
     this._engine = newEngine();
   }
 
+  get _document() {
+    return (async () => (await this._subject).replace(/#.*/, ''))();
```

Comment:
This is quite a bit more complicated 🙂

Reply:
Yeah, getters can't be async unfortunately, so we have to work around that :/

Reply:
alternatively: `getDocument(subject)`

Reply:
Sure, that works as well, whatever you prefer :-)

Reply:
Will be more legible, so I'd recommend that.